code/src/containers/forms/new-product-form/NewProductForm.test.tsx

```diff
@@ -0,0 +1,328 @@
+
+    expect(nameInput.closest(".Mui-error")).toBeTruthy();
+    expect(descriptionInput.closest(".Mui-error")).toBeTruthy();
+    expect(imgUrlInput.closest(".Mui-error")).toBeTruthy();
+    expect(priceInput.closest(".Mui-error")).toBeTruthy();
+    expect(quantityInput.closest(".Mui-error")).toBeTruthy();
+  });
+
+  test("Should display message to fill in at least one translation, if none was filled", async () => {
+    render();
+
+    await typeIntoInput(imgUrlInput, testData.image);
+    await typeIntoInput(priceInput, testData.price);
+    await typeIntoInput(quantityInput, testData.quantity);
+
+    await selectCategory();
+    await submit();
+
+    expect(mockShowSnackbar).toHaveBeenCalledWith({
+      variant: "error",
+      messageTranslationKey: "productForm.creation.translationError"
+    });
+  });
+
+  test("Should show success message, redirect and send request if validation was passed", async () => {
+    render();
+
+    await fillInTestData();
+    await selectCategory();
+    await submit();
+
+    expect(mockCreateProductMutation).toHaveBeenCalledWith(testData);
+    expect(mockShowSnackbar).toHaveBeenCalledWith({
+      variant: "success",
+      messageTranslationKey: "productForm.creation.success"
+    });
+    expect(mockNavigate).toHaveBeenCalledWith(routes.dashboard.products.path);
+  });
+
+  test("Should change product status and send request with selected status", async () => {
+    render();
+
+    await act(async () => {
+      fireEvent.click(statusInput);
+    });
+
+    await fillInTestData();
+    await selectCategory();
+    await submit();
+
+    expect(mockCreateProductMutation).toHaveBeenCalledWith({
+      ...testData,
+      status: "VISIBLE"
+    });
+    expect(mockShowSnackbar).toHaveBeenCalledWith({
+      variant: "success",
+      messageTranslationKey: "productForm.creation.success"
+    });
+    expect(mockNavigate).toHaveBeenCalledWith(routes.dashboard.products.path);
+  });
+
+  test("Should send request with 2 translations if it was added by user", async () => {
+    render();
+
+    await fillInTestData();
+    await selectCategory();
+
+    const nameInputUk = screen
+      .getByTestId(`new-product-name-input-uk`)
+      .querySelector("input")!;
+    const descriptionInputUk = screen
+      .getByTestId(`new-product-description-input-uk`)
+      .querySelector("textarea")!;
+
+    await typeIntoInput(nameInputUk, testData.productTranslations[0].name);
+    await typeIntoInput(
+      descriptionInputUk,
+      testData.productTranslations[0].description
+    );
+
+    await submit();
+
+    expect(mockCreateProductMutation).toHaveBeenCalledWith(
+      twoTranslationsTestData
+    );
+  });
+
+  test("Should show default error if the request error is not related to validation", async () => {
+    mockUnwrap.mockImplementation(() => {
+      throw { status: 500 };
+    });
+
+    render();
+
+    await fillInTestData();
+    await selectCategory();
+    await submit();
```

Comment:
These three lines are repeated. I suggest creating nested `describe` block and move repeated logic to `beforeEach` or setup-function

Reply:
Thanks, I made a function `baseSteps` and used it in test cases where it is needed.